lib/netstd/Thrift/Transport/Client/THttpTransport.cs

```diff
@@ -57,6 +57,27 @@ public THttpTransport(Uri uri, TConfiguration config, IDictionary<string, string
             // due to current bug with performance of Dispose in netcore https://github.com/dotnet/corefx/issues/8809
             // this can be switched to default way (create client->use->dispose per flush) later
             _httpClient = CreateClient(customRequestHeaders);
+            ConfigureClient(_httpClient);
+        }
+
+        /// <summary>
+        /// Constructor that takes a <c>HttpClient</c> instance to support using <c>IHttpClientFactory</c>.
+        /// </summary>
+        /// <remarks>As the <c>HttpMessageHandler</c> of the client must be configured at the time of creation, it
+        /// is assumed that the consumer has already added any certificates and configured decompression methods.</remarks>
```

Comment:
Should we create a public static method like this:

```csharp
public static HttpMessageHandler (IEnumerable<X509Certificate> certificates)
{
    var handler = new HttpClientHandler();
    handler.ClientCertificates.AddRange(_certificates);
    handler.AutomaticDecompression = System.Net.DecompressionMethods.Deflate | System.Net.DecompressionMethods.GZip;
    return handler;
}
```

That way the user wouldn't need to dig into the source code of this method to figure out which decompression methods are typically configured, and they could easily load in any certificates. We could then use the same method inside the `CreateClient()` method to stop state drifting.